sdks/python/apache_beam/io/bigquery.py

```diff
@@ -688,6 +723,21 @@ def _delete_table(self, project_id, dataset_id, table_id):
         projectId=project_id, datasetId=dataset_id, tableId=table_id)
     self.client.tables.Delete(request)
 
+  @retry.with_exponential_backoff()  # Using retry defaults from utils/retry.py
+  def _delete_dataset(self, project_id, dataset_id, delete_contents=True):
+    request = bigquery.BigqueryDatasetsDeleteRequest(
+        projectId=project_id, datasetId=dataset_id,
+        deleteContents=delete_contents)
+    self.client.datasets.Delete(request)
+
+  def create_temporary_dataset(self, project_id):
+    dataset_id = BigQueryWrapper.TEMP_DATASET + self._temporary_table_suffix
+    self._create_dataset(project_id, dataset_id)
+
+  def clean_up_temporary_dataset(self, project_id):
+    temp_table = self._get_temp_table(project_id)
```

Comment:
Check if the dataset is not available and print a warning instead of the following call if dataset is not available.